### PR TITLE
[typescript][node]: Add accept header if produces is not empty

### PR DESCRIPTION
**Pull request by clemens-smartparking**
Uses the correct Accept media types as specified in the OpenApi
specifications.

relates to #3944

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.
@akehir @macjohnny
### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache [hunk 1]
```diff
@@ -134,6 +134,9 @@ export class {{classname}} {
             .replace('{' + '{{baseName}}' + '}', encodeURIComponent(String({{paramName}}))){{/pathParams}};
         let localVarQueryParameters: any = {};
         let localVarHeaderParams: any = (<any>Object).assign({}, this.defaultHeaders);
+{{#hasProduces}}
+        localVarHeaderParams.Accept = {{#produces}}{{#-first}}'{{{mediaType}}}'{{/-first}}{{/produces}};
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache]: why only the first one?

File: samples/client/petstore/typescript-node/default/api/petApi.ts [hunk 2]
```diff
@@ -379,17 +382,17 @@ export class PetApi {
     /**
      * 
      * @summary Update an existing pet
-     * @param body Pet object that needs to be added to the store
+     * @param pet Pet object that needs to be added to the store
      */
-    public async updatePet (body: Pet, options: {headers: {[name: string]: string}} = {headers: {}}) : Promise<{ response: http.ClientResponse; body?: any;  }> {
+    public async updatePet (pet: Pet, options: {headers: {[name: string]: string}} = {headers: {}}) : Promise<{ response: http.ClientResponse; body?: any;  }> {
```
Comment (macjohnny) [inline samples/client/petstore/typescript-node/default/api/petApi.ts]: please run the [bin/typescript-node-petstore-all.sh](https://github.com/OpenAPITools/openapi-generator/blob/master/bin/typescript-node-petstore-all.sh) script only, not the `openapi3` version

**Comment by macjohnny** [comment]
thanks for the PR

**Comment by clemens-smartparking** [comment]
Thanks for the review! Somehow managed to copied the {{#-first}} bit from the lua code, the only generator who seems to do it differently... I updated the pull request and the samples look much better now.

**Comment by macjohnny** [comment]
the tests seem to fail, can you have a look at https://travis-ci.org/OpenAPITools/openapi-generator/builds/591210342?utm_source=github_status&utm_medium=notification ?

**Comment by clemens-smartparking** [comment]
After running the client.ts through a debugger I found out that getPetById specifies to produce 'application/xml' and 'application/json'. When accepting both (in that order), the endpoint returns xml. That fails to deserialize...

What to you reckon? Never include 'application/xml'? If 'application/json' is present, don't include anything else?

And how should that logic be implemented? There is no `if` in mustache, so how to modify the `produces` field?

**Comment by macjohnny** [comment]
thanks for the analysis, good to know.
see e.g.
https://github.com/OpenAPITools/openapi-generator/blob/c8ac41c051f0557e5a1d22ad2737bb3cde16b650/samples/client/petstore/typescript-angular-v7-provided-in-root/builds/default/api/pet.service.ts#L89-L106
https://github.com/OpenAPITools/openapi-generator/blob/c8ac41c051f0557e5a1d22ad2737bb3cde16b650/modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache#L250-L264

**Comment by clemens-smartparking** [comment]
Ok good idea, would you be happy with this solution:
```
{{#hasProduces}}
        const produces = [{{#produces}}'{{{mediaType}}}'{{#hasMore}}, {{/hasMore}}{{/produces}}];
        // give precedence to 'application/json'
        if (produces.indexOf('application/json') >= 0) {
            localVarHeaderParams.Accept = 'application/json';
        } else {
            localVarHeaderParams.Accept = produces.join(',');
        }
{{/hasProduces}}
```

**Pull request merged**